config/v1/types_authentication.go

```diff
@@ -243,11 +244,30 @@ type OIDCProvider struct {
 	// +listType=atomic
 	// +optional
 	ClaimValidationRules []TokenClaimValidationRule `json:"claimValidationRules,omitempty"`
+
+	// UserValidationRules defines the set of rules used to validate claims in a user's token.
+	// Each rule is evaluated independently to determine whether the token subject is considered valid.
+	// Rules can either require specific claims and values to be present,
+	// or define CEL expressions that must evaluate to true for the token to be accepted.
+	// If the expression in a rule evaluates to false, the token is rejected.
+	// At least one rule must evaluate to true for the token to be considered valid.
+	// A maximum of 64 rules can be specified. This field is optional.
+	//
+	// See https://kubernetes.io/docs/reference/using-api/cel/ for CEL syntax.
+	// +listType=atomic
+	// +kubebuilder:validation:MaxItems=64
+	// +kubebuilder:validation:XValidation:rule="self.size() == 0 || <feature-gate-enabled-check>",message="user validation rules are not supported when StructuredAuthenticationConfiguration feature gate is disabled"
+	// +kubebuilder:validation:XValidation:rule="self.all(x, x.expression.size() > 0)",message="each expression must be non-empty"
```

Comment:
Expression is already required with a minimum length of 1, meaning it cannot be empty. That should be sufficient enforcement here, no need for this CEL check as well.